applications/MeshingApplication/custom_utilities/uniform_refine_utility.h

```diff
@@ -218,8 +218,8 @@ class UniformRefineUtility
     std::unordered_map<IndexType,int> mNodesColorMap;
     std::unordered_map<IndexType,int> mCondColorMap;
     std::unordered_map<IndexType,int> mElemColorMap;
-    std::unordered_map<int,std::vector<std::string>> mColors;  /// Where the sub model parts IDs are stored
-    std::map<std::pair<int,int>,int> mIntersections;           /// Where the colors intersection are stored
+    std::unordered_map<int,std::vector<ModelPart*>> mColorsPointers; /// Where the sub model parts are stored
```

Comment:
Be careful with the raw pointers!!!!, the raw pointers are evil (potential memory leak)

Reply:
I recommend to use smart pointers if possible

Reply:
I want to use ModelPart::Pointer but nowadays I think it is not possible...

Reply:
Let me check...